### PR TITLE
Create pgtypes (and slices of) from pointers

### LINKED ISSUE #40: Support slice of pointers
We had a similar problem to [pgx#242](https://github.com/jackc/pgx/issues/242), however in our case we needed a `[]*time.Time`.

We have to call a Postgres function to insert data which takes a couple arrays as input parameters. One of which is a an ETA defined as: `eta_ TIMESTAMP WITH TIME ZONE[]`. In go we have a `var etas []*time.Time` where some of the elements are `nil` (as the ETA is not yet known) and those should be stored as `NULL` in our database.

I propose a solution to solve this particular issue with `[]*time.Time` which works with pgx v4 out of the box, see #41. If you find this good quality I'd be pleased to make the fix for every array types. I'd also appreciate if you had a smarter way of dealing with this issue since there's a bit of code repetition.

### PR DESCRIPTION
**Pull request by megaturbo**
Fixes #40

### CODE REVIEW

File: convert.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func underlyingTimeType(val interface{}) (interface{}, bool) {
 	switch refVal.Kind() {
 	case reflect.Ptr:
 		if refVal.IsNil() {
-			return nil, false
+			return nil, true
```
Comment (jackc) [inline convert.go:152]: I think this would allow any type of nil, not just a `*time.Time`. Not sure if that's really a problem.

Comment (megaturbo) [reply convert.go:152]: That was a mistake, this won't stay

**Comment by megaturbo** [comment]
I took inspiration on how to solve this where you already faced this issue on [`aclitem.go`](https://github.com/jackc/pgtype/blob/master/aclitem.go#L42) for example. This is not DRY at all, but I cannot think of a better way using type switches...

For the array types I did the changes through `typed_array_gen.sh` and added the pointer types as you suggested.

Again if you have a better idea on how to solve this issue I'd be very happy to implement it.

**Comment by megaturbo** [comment]
@jackc Tell me what you think about these changes. I consider adding tests for those, current tests are passing but I don't think they're relevant to the changes I made. However it works on my project where I needed this feature.

PS: I took the liberty to add `Code generated by...` notice at the top of the file. (Easier git diff through github and my IDE likes it too)

**Comment by megaturbo** [comment]
Sorry I forgot `[]*string`, it's fixed now

**Comment by jackc** [comment]
LGTM. I don't think there is any way around the verbosity. Maybe someday it can be streamlined if Go generics ever happens.

**Pull request merged**